File: src/main/java/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/finder/Finder.java
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.finder;

import io.github.oleksivio.telegram.bot.core.controller.handler.check.Validator;
import io.github.oleksivio.telegram.bot.core.model.ITelegram;

import java.lang.annotation.Annotation;

public interface Finder {
    <SA extends Annotation, T> Validator<T> find(SA annotation);

    <T extends ITelegram> Validator<T> find(String validatorName, Class<T> targetClass);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/finder/Finder.kt
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.finder

import io.github.oleksivio.telegram.bot.core.controller.handler.check.Validator
import io.github.oleksivio.tl.kbot.server.api.model.ITelegram
import kotlin.reflect.KClass

interface Finder {
    fun <SA : Annotation, T> find(annotation: SA): Validator<T>

    fun <T : ITelegram> find(validatorName: String, targetClass: KClass<T>): Validator<T>

}
